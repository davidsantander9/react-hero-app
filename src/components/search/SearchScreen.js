import React from 'react'
import { heroes } from '../../data/heroes'
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../heroes/HeroCard';

export const SearchScreen = () => {

    const heroesFiltered = heroes;

    const [ { search }, handleInputChange, reset ] = useForm({
        search: '',
    });

    const handleSearch = (e) => {
        e.preventDefault();

        if (search.trim() <= 1) {
            return
        }
        
        reset();
    }
    return (
        <div>
            <h1>Search Screen</h1>
            <div className="row">
                <div className="col-5">
                    <h4>Search Form</h4>
                    <hr/>
                    <form onSubmit={ handleSearch }>
                        <input
                            type="text"
                            name="search"
                            autoComplete="off"
                            placeholder="Find your hero"
                            className="form-control"
                            value={search}
                            onChange={ handleInputChange }
                        />
                        <button
                            type="submit"
                            className="btn m-1 btn-block btn-outline-primary"
                        >
                            Search ...
                        </button>
                    </form>
                </div>
                <div className="col-7">
                    <h4>Result</h4>
                    <hr/>
                    {
                        heroesFiltered.map( hero => (
                            <HeroCard
                                key={ heroes.id}
                                {...hero}
                            />
                        ))
                    }
                </div>
            </div> 
        </div>
    )
}
