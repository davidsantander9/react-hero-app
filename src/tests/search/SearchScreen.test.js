import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../components/search/SearchScreen';

describe('Test on <SearchScreen/>', () => {

    test('should show with default values', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        );

        expect( wrapper ).toMatchSnapshot();
        // expect( wrapper.find('.alert-info').text().trim() ).toBe();

    });

    test('should show batman and the query-string value', () => {
       
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=spider']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        );

        // expect( wrapper.find('input').prop('value') ).toBe('spider');
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show batman and the query-string value', () => {
       
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=spider']}>
                <Route path="/search" component={ SearchScreen }/>
            </MemoryRouter>
        );

        // expect( wrapper.find('input').prop('value') ).toBe('spider');
        expect( wrapper ).toMatchSnapshot();

    });

    // test('should call history push', () => {
        
    //     const wrapper = mount(
    //         <MemoryRouter initialEntries={['/search?q=spider']}>
    //             <Route path="/search" component={ SearchScreen }/>
    //         </MemoryRouter>
    //     );


    // })
    
    

    
})
