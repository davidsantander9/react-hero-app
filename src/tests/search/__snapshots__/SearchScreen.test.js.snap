// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <SearchScreen/> should show batman and the query-string value 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="search"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search ...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <a
        className="my-card"
        href="/hero/marvel-spider"
        onClick={[Function]}
      >
        <img
          alt="Spider Man"
          className="img img-responsive"
          src="./assets/heroes/marvel-spider.jpg"
        />
        <div
          className="profile-name"
        >
          Spider Man
        </div>
        <div
          className="profile-position"
        >
          Peter Parker
        </div>
        <div
          className="profile-overview"
        >
          <div
            className="profile-overview"
          >
            <div
              className="row"
            >
              <div
                className="col-ms-4"
              >
                <h3>
                  Marvel Comics
                </h3>
                <p>
                  Primera aparici√≥n: 
                  <br />
                  Amazing Fantasy #15
                </p>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Test on <SearchScreen/> should show with default values 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="search"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search ...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
    </div>
  </div>
</div>
`;
